modules/openapi-generator/src/main/resources/cpp-pistache-server/modelbase-header.mustache

```diff
@@ -39,100 +39,85 @@ public:
     static nlohmann::json toJson({{prefix}}ModelBase const& content ); 
 };
 
-class {{prefix}}ModelArrayHelper {
-public:
-	template<typename T>
-	static std::vector<T> fromJson(nlohmann::json& json) {
-		T *ptrTest;
-		std::vector<T> val;
-		if (dynamic_cast<{{prefix}}ModelBase*>(ptrTest) != nullptr) {
-			if (!json.empty()) {
-				for (auto &item : json.items()) {
-					T entry;
-					entry.fromJson(item.value());
-					val.push_back(entry);
-				}
-			}
-		}
-		return val;
-	}
-	template<typename T>
-	static nlohmann::json toJson(std::vector<T> val) {
-		nlohmann::json json;
-		for(auto item : val){
-			json.push_back(item.toJson());
-		}
-		return json;
-	}
-};
-
 class {{prefix}}ArrayHelper {
+private:
+    template<typename T, typename std::enable_if<!std::is_base_of<ModelBase, T>::value>::value>
+    static void itemFromJson(T &item, nlohmann::json &json){
+        item = json;
+    }
+    static void itemFromJson(ModelBase &item, nlohmann::json &json){
+        item.fromJson(json);
+    }
+    template<typename T, typename std::enable_if<!std::is_base_of<ModelBase, T>::value>::value>
+    static nlohmann::json itemtoJson(T &item){
```

Comment:
is the object going to change here? if not make the parameter const.

Reply:
Done